actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -897,7 +897,8 @@ def connect(*path_or_actions, as: DEFAULT, to: nil, controller: nil, action: nil
             assign_deprecated_options(deprecated_options, mapping, :connect)
           end
 
-          match(*path_or_actions, as:, to:, controller:, action:, on:, defaults:, constraints:, anchor:, format:, path:, internal:, **mapping, via: [:get, :connect], &block)
+          via = [:get, :connect]
```

Comment:
This one is a bit surprising. The array being a literal I'd expect it not to be subject to that problem.

Reply:
This is probably possible to fix in Ruby's compilers.  You'd need to inspect all subnodes of NODE_LIST/ArrayNode and check if all of them are also safe.  Those nodes are not currently handled, so the default is to treat them as unsafe.

Reply:
Submitted a pull request to Ruby to avoid array allocation for literal arrays if all elements of the array are safe (which they are in this case): https://github.com/ruby/ruby/pull/13673.  With the optimization, you would not need to create a local variable to avoid the allocation.